providers/oidc.go

```diff
@@ -170,7 +164,16 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	newSession.AccessToken = token.AccessToken
 	newSession.RefreshToken = token.RefreshToken
 	newSession.CreatedAt = &created
-	newSession.ExpiresOn = &token.Expiry
+
+	var tokenExpiry time.Time
+	if idToken == nil || token.Expiry.Before(idToken.Expiry) {
```

Comment:
NIT: I'm partial to formatting like this to reduce the amount of branching in the code:

```
tokenExpiry := token.Expiry
if idToken != nil && token.Expiry.After(idToken.Expiry) {
	tokenExpiry = idToken.Expiry
}
```

Reply:
That's fine.